pkg/errors/error.go

```diff
@@ -50,22 +50,35 @@ const (
 	errDatabase     = errClass("database")
 )
 
+func isCodeCancellation(code gcodes.Code) bool {
+	return code == gcodes.Canceled || code == gcodes.DeadlineExceeded
+}
+
+func isErrorCancellation(err error) bool {
+	if errors.Is(err, context.Canceled) || errors.Is(err, context.DeadlineExceeded) {
```

Comment:
Same here, we could use an array and iterate, but kept it like this since we only compare to 2 values